### DISCUSSION TITLE
How can I tell which runtime the package can be used on?

### DISCUSSION THREAD

**Discussion by sz3lbi** [gleam-lang/gleam#3589] [Questions & support] [answered]
Hello,
I am starting my journey with Gleam and I am trying to understand the whole environment around it.
Since the language itself can be compiled for the Erlang or JavaScript runtime, the packages I want to use should also be targeted for that specific runtime. However, the runtime that a specific package targets is not so obvious to me.

There are several cases:
1. Excellent examples of packages with repository badges such as: [glenvy](https://github.com/maxdeviant/glenvy), [birdie](https://github.com/giacomocavalieri/birdie), [go_over](https://github.com/bwireman/go-over)
Just open the docs or the repository and you know straight away which targets are supported!
2. Packages with descriptions like: [simplifile](https://github.com/bcpeinhardt/simplifile), [filepath](https://github.com/lpil/filepath)
You open the docs or the repository, you read the description for a few seconds, and you know which targets are supported!
3. Packages that target the JavaScript runtime, but don't say if it's for the browser or maybe Node/Deno/Bun.
4. Packages with no information in this regard.

How do I handle cases 3 & 4 and know which runtime they are targeting? Of course, I can just guess, and for packages that use the [gleam_erlang](https://github.com/gleam-lang/erlang) dependency I can tell, but there are less obvious ones. Maybe the badge approach should just become a standard?

Thanks in advance for your explanation.

**Comment by lpil** [comment, marked as answer]
Hello! Today you need to read the README, but there is work in place to infer and surface target support on the packages site.

We will never be able to infer runtime support so it'll be up for the library authors to decide what (if any) runtimes they want to explicitly support and document this accordingly. This is the case for all programming languages unfortunately.

**Reply by sz3lbi** [reply]
Is there an issue or pull request I can track to see progress on the functionality you mentioned?

**Reply by giacomocavalieri** [reply]
I couldn't find one, so I've opened an issue you can track here <https://github.com/gleam-lang/gleam/issues/3684> :)

**Reply by sz3lbi** [reply]
Thanks @lpil and @giacomocavalieri!
